samples/drivers/i2s/i2s_codec/README.rst

```diff
@@ -0,0 +1,33 @@
+.. zephyr:code-sample:: i2s_codec
+   :name: I2S codec
+   :relevant-api: i2s_interface
+
+   Process an audio stream to codec.
+
+Overview
+********
+
+This sample demonstrates how to use an I2S driver in a simple processing of
+an audio stream. It configures and starts from memory buffer or from DMIC to
+record i2s data and send to codec(on board WM8904) with DMA.
```

Comment:
the general overview should not mention anything hardware specific (hopefully the sample works with other codecs than WM8904? :) )

Reply:
this sample application requires board to have dmic and codec, so I did not have chance test on other platform, basically they should working but I do not know which platform can have the same